src/ray/core_worker/core_worker.h

```diff
@@ -62,60 +62,218 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
+/// associated to the only worker. Then no need to call `SetCurrentThreadWorkerId` in
+/// your own threads. Currently a Python worker process starts only 1 worker.
+class CoreWorkerProcess {
+ public:
+  ///
+  /// Public methods used in both DRIVER and WORKER mode.
+  ///
+
+  /// Initialize core workers at process level.
+  ///
+  /// \param[in] options The various initialization options.
+  static void Initialize(const CoreWorkerOptions &options);
+
+  /// Get the core worker associated with the current thread.
+  /// NOTE (kfstorm): Here we return a reference instead of a `shared_ptr` to make sure
+  /// `CoreWorkerProcess` has full control of the destruction timing of `CoreWorker`.
+  static CoreWorker &GetCoreWorker();
+
+  /// Set the core worker associated with the current thread by worker ID.
+  /// Currently used by Java worker only.
+  ///
+  /// \param worker_id The worker ID of the core worker instance.
+  static void SetCurrentThreadWorkerId(const WorkerID &worker_id);
+
+  /// Where the current process has initialized for core worker.
+  static bool IsInitialized();
+
+  ///
+  /// Public methods used in DRIVER mode only.
+  ///
+
+  /// Shutdown workers completely at process level.
+  /// This API is for driver only.
+  static void Shutdown();
+
+  ///
+  /// Public methods used in WORKER mode only.
+  ///
+
+  /// Start receiving and executing tasks.
+  /// This API is for worker only.
+  static void StartExecutingTasks();
+
+  // The destructor is not a public API to use. However it's required by smart pointer.
+  ~CoreWorkerProcess();
+
+ private:
+  /// Create an `CoreWorkerProcess` with proper options.
+  ///
+  /// \param[in] options The various initialization options.
+  CoreWorkerProcess(const CoreWorkerOptions &options);
+
+  /// Check that the core worker environment is initialized for this process.
+  ///
+  /// \return Void.
+  static void EnsureInitialized();
+
+  /// Get the `CoreWorker` instance by worker ID.
+  ///
+  /// \param[in] workerId The worker ID.
+  /// \return The `CoreWorker` instance.
+  std::shared_ptr<CoreWorker> GetWorker(const WorkerID &worker_id) const
+      LOCKS_EXCLUDED(worker_map_mutex_);
+
+  /// Create a new `CoreWorker` instance.
+  ///
+  /// \return The newly created `CoreWorker` instance.
+  std::shared_ptr<CoreWorker> CreateWorker() LOCKS_EXCLUDED(worker_map_mutex_);
+
+  /// Remove an existing `CoreWorker` instance.
+  ///
+  /// \param[in] The existing `CoreWorker` instance.
+  /// \return Void.
+  void RemoveWorker(std::shared_ptr<CoreWorker> worker) LOCKS_EXCLUDED(worker_map_mutex_);
+
+  /// The global instance of `CoreWorkerProcess`.
+  static std::unique_ptr<CoreWorkerProcess> instance_;
+
+  /// The various options.
+  const CoreWorkerOptions options_;
+
+  /// The core worker instance associated with the current thread.
+  /// Use weak_ptr here to avoid memory leak due to multi-threading.
```

Comment:
The task execution thread cannot outlive the `CoreWorker` instance, but the user-created threads can. Especially that it's common in Java to use static methods to execute something asynchronously, which means using the global thread pool instead of creating your own. Besides, users can still run other kinds of stuff in the main thread after the driver is shut down, right?